code/modules/power/singularity/field_generator.dm

```diff
@@ -344,10 +344,10 @@ field_generator power level display
 	//I want to avoid using global variables.
 	spawn(1)
 		var/temp = 1 //stops spam
-		for(var/obj/singularity/O in machines)
+		for(var/obj/singularity/O in world)

```

Comment:
why would you ever change this to an `in world` call
make a new global list for singularities or just keep adding them to machines, don't use `in world`, that's horrid